ui-common/src/main/java/io/syndesis/qe/steps/CommonSteps.java

```diff
@@ -933,7 +944,23 @@ private void loginToGoogleIfNeeded(String s) {
         }
 
         waitForCallbackRedirect("google");
-        fillAndValidateGoogleAccount(s);
+
+        Account account = AccountsDirectory.getInstance().get(s);
+        // test if multi accounts table is shown ( https://accounts.google.com/accountchooser )
+        SelenideElement multiAccountsTable = $(By.className("OVnw0d"));
+        if (multiAccountsTable.exists()) {
+            log.info("More Google accounts are available. Choosing the correct one:" + account.getProperty("email"));
+            try {
+                multiAccountsTable.find(ByUtils.customAttribute("data-email", account.getProperty("email"))).should(Condition.exist).click();
+            } catch (ElementNotFound ex) {
+                log.warn("That account is not in the Google multi-accounts table. It needs to fill in the credentials.");
+                $(By.className("UXurCe")).click(); // use another account button
```

Comment:
The question is how to implement By.text method since parents contains that text too (first 4 parent element has only that text too so "exact match" still returns 4 elements). Hopefully the root parent with the same text is clickable :D